test/unit/org/apache/cassandra/cql3/validation/entities/UFTest.java

```diff
@@ -1068,4 +1070,26 @@ public void testRejectInvalidLanguageOnCreation()
                                   "For more information - CASSANDRA-18252");
         }
     }
+
+    @Test
+    public void testFunctionsAreNonMonotonicAndNonDeterministicByDefault() throws Throwable
+    {
+        createFunction(KEYSPACE, "int", "CREATE FUNCTION %s (x int) " +
+                "CALLED ON NULL INPUT " +
+                "RETURNS int " +
+                "LANGUAGE java " +
+                "AS 'return x / 2;'");
+
+        UntypedResultSet functions = execute("SELECT * FROM system_schema.functions");
+
+        Assert.assertEquals(1, functions.size());
+        Assert.assertFalse(functions.one().getBoolean("monotonic"));
+        Assert.assertFalse(functions.one().getBoolean("deterministic"));
+        Assert.assertEquals(Collections.emptyList(), functions.one().getList("monotonic_on", UTF8Type.instance));
+
+        UntypedResultSet aggregates = execute("SELECT * FROM system_schema.aggregates");
+
+        Assert.assertEquals(1, aggregates.size());
```

Comment:
We haven't created any aggregation function in this test method, so this will fail.

Reply:
Also, this seems to do mostly the same as the new `AggregationTest#testAggregatesAreNonDeterministicByDefault`.

Reply:
For a moment I was tempted to remove the `UFTest#testFunctionsAreNonMonotonicAndNonDeterministicByDefault` and leave only `AggregationTest#testAggregatesAreNonDeterministicByDefault` since the latter has to create the function in order to later create the aggregation. Eventually, I decided to leave both since I didn't want to assert on the function-specific columns in the `AggregationTest` but I am not calling that an ultimate solution - if you think it's cleaner to have these two merged into one I am fine with doing it too.